dev-util/bear/bear-3.0.10.ebuild

```diff
@@ -0,0 +1,81 @@
+# Copyright 2020-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{7..9} )
+
+inherit cmake python-single-r1
+
+DESCRIPTION="Build EAR generates a compilation database for clang tooling"
+HOMEPAGE="https://github.com/rizsotto/Bear"
+SRC_URI="https://github.com/rizsotto/Bear/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-3+"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc64"
+IUSE="test"
+
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+RDEPEND="
+	${PYTHON_DEPS}
+	>=dev-cpp/nlohmann_json-3.7
+	>=dev-db/sqlite-3.14
+	>=dev-libs/libfmt-6.2
+	>=dev-libs/spdlog-1.5
+	>=net-libs/grpc-1.26
+"
+
+DEPEND="${RDEPEND}
+test? (
+	>=dev-cpp/gtest-1.10
+)"
+
+BDEPEND="test? (
+	$(python_gen_cond_dep '
+		dev-python/lit[${PYTHON_USEDEP}]
+	')
+)"
+
+RESTRICT="!test? ( test )"
+
+S="${WORKDIR}/${P^}"
+
+src_configure() {
+	local mycmakeargs=(
+		"-DENABLE_UNIT_TESTS=OFF"
```

Comment:
no, because then we'd build and execute tests during src_compile. bears cmake config has no way (that I found) of running tests seperately

Reply:
I think he meant $(usex test on off). This will check if actually test is required or not and add/remove it as arg

Reply:
Bear has tests on by default, which is why I disable them in configure and reenable them in test

Reply:
I think if you change `TEST_BEFORE_INSTALL` to `0` it will be possible to skip tests, can you try and test it?

https://github.com/rizsotto/Bear/blob/d84687e2112e48263ebe18ed022a36c975c3ccf6/CMakeLists.txt#L63-L64

also it should be possible to run tests separately with `ctest --verbose || die`

Reply:
doesn't work, both test projects are still run during src_compile